weave/trace_server/clickhouse_trace_server_batched.py

```diff
@@ -185,6 +185,8 @@ def __init__(
     ):
         super().__init__()
         self._thread_local = threading.local()
+        self._secondary_thread_local = threading.local()
```

Comment:
idk if this is the best solution as it hard codes a depth of 2 into the design. It seems like instead you want a design that allows the caller to say "with new_client():" and then calls underneath that could do the same.